python/tvm/driver/tvmc/runner.py

```diff
@@ -0,0 +1,464 @@
+        Shape dictionary - {input_name: tuple}.
+    dtype_dict : dict
+        dtype dictionary - {input_name: dtype}.
+    fill_mode : str
+        The fill-mode to use when generating tensor data.
+        Can be either "zeros", "ones" or "random".
+
+    Returns
+    -------
+    inputs_dict : dict
+        Complete inputs dictionary - {input_name: np.array}.
+    """
+    logger.debug("creating inputs dict")
+
+    try:
+        inputs = np.load(inputs_file) if inputs_file else {}
+    except IOError as ex:
+        raise TVMCException("Error loading inputs file: %s" % ex)
+
+    # First check all the keys in inputs exist in the graph
+    for input_name in inputs:
+        if input_name not in shape_dict.keys():
+            raise TVMCException(
+                "the input tensor '{}' is not in the graph. Expected inputs: '{}'".format(
+                    input_name, shape_dict.keys()
+                )
+            )
+
+    # Now construct the input dict, generating tensors where no
+    # data already exists in 'inputs'
+    inputs_dict = {}
+    for input_name in shape_dict:
+        if input_name in inputs.keys():
+            logger.debug("setting input '%s' with user input data", input_name)
+            inputs_dict[input_name] = inputs[input_name]
+        else:
+            shape = shape_dict[input_name]
+            dtype = dtype_dict[input_name]
+
+            logger.debug(
+                "generating data for input '%s' (shape: %s, dtype: %s), using fill-mode '%s'",
+                input_name,
+                shape,
+                dtype,
+                fill_mode,
+            )
+            data = generate_tensor_data(shape, dtype, fill_mode)
+            inputs_dict[input_name] = data
+
+    return inputs_dict
+
+
+def run_module(
+    module_file,
+    hostname,
+    port=9090,
+    rpc_key=None,
+    device=None,
+    inputs_file=None,
+    fill_mode="random",
+    repeat=1,
+    profile=False,
+):
+    """Run a compiled graph runtime module locally or remotely with
+    optional input values.
+
+    If input tensors are not specified explicitly, they can be filled
+    with zeroes, ones or random data.
+
+    Parameters
+    ----------
+    module_file : str
+        The path to the module file (a .tar file).
+    hostname : str
+        The hostname of the target device on which to run.
+    port : int, optional
+        The port of the target device on which to run.
+    rpc_key : str, optional
+        The tracker key of the target device. If this is set, it
+        will be assumed that remote points to a tracker.
+    device: str, optional
+        the device (e.g. "cpu" or "gpu") to be targeted by the RPC
+        session, local or remote).
+    inputs_file : str, optional
+        Path to an .npz file containing the inputs.
+    fill_mode : str, optional
+        The fill-mode to use when generating data for input tensors.
+        Valid options are "zeros", "ones" and "random".
+        Defaults to "zeros".
```

Comment:
random.

Reply:
I'll fix that in a sec.

Reply:
https://github.com/apache/incubator-tvm/pull/6578/commits/80c36da00a8cea289ca562e7d9d87d23bedb578e